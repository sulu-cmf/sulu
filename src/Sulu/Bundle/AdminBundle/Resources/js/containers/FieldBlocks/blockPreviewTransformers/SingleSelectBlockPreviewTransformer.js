// @flow
import React from 'react';
import type {Node} from 'react';
import type {BlockPreviewTransformer} from '../types';
import type {SchemaEntry} from '../../Form/types';

export default class SingleSelectBlockPreviewTransformer implements BlockPreviewTransformer {
    transform(value: *, schema: SchemaEntry): Node {
        if (!schema.options || !schema.options.values) {
            throw new Error('The "SingleSelect" field type must have a "values" schema option!');
        }

        const values = schema.options.values.value;
        if (!Array.isArray(values)) {
            throw new Error('The "SingleSelect" field type must have a "values" option defined being an array!');
        }

        const selectedValue = values.find((option) => option.name === value);

        if (!selectedValue) {
            return null;
        }

        return <p>{selectedValue.title}</p>;
    }
}
