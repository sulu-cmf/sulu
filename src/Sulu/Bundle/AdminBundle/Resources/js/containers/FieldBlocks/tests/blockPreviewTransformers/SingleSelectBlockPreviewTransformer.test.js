// @flow
import SingleSelectBlockPreviewTransformer from '../../blockPreviewTransformers/SingleSelectBlockPreviewTransformer';

test('Return JSX for selected item', () => {
    const singleSelectBlockPreviewTransformer = new SingleSelectBlockPreviewTransformer();
    expect(
        singleSelectBlockPreviewTransformer.transform(
            'value1',
            {
                options: {
                    values: {
                        value: [
                            {
                                name: 'value1',
                                title: 'Value 1',
                            },
                            {
                                name: 'value2',
                                title: 'Value 2',
                            },
                            {
                                name: 'value3',
                                title: 'Value 3',
                            },
                        ],
                    },
                },
                type: 'single_select',
            }
        )
    ).toMatchSnapshot();
});

test('Return null if nothing is passed', () => {
    const singleSelectBlockPreviewTransformer = new SingleSelectBlockPreviewTransformer();
    expect(
        singleSelectBlockPreviewTransformer.transform(
            undefined,
            {
                options: {
                    values: {
                        value: [
                            {
                                name: 'value1',
                                title: 'Value 1',
                            },
                            {
                                name: 'value2',
                                title: 'Value 2',
                            },
                            {
                                name: 'value3',
                                title: 'Value 3',
                            },
                        ],
                    },
                },
                type: 'single_select',
            }
        )
    ).toMatchSnapshot();
});

test('Throw an error if schema has no values schema option', () => {
    const singleSelectBlockPreviewTransformer = new SingleSelectBlockPreviewTransformer();
    expect(
        () => singleSelectBlockPreviewTransformer.transform('value1', {type: 'single_select'})
    ).toThrow(/"values" schema option/);
});

test('Throw an error if values schema option is not an array', () => {
    const singleSelectBlockPreviewTransformer = new SingleSelectBlockPreviewTransformer();
    expect(
        () => singleSelectBlockPreviewTransformer.transform(
            'value3',
            {
                options: {
                    values: {},
                },
                type: 'single_select',
            }
        )
    ).toThrow(/"values" option defined being an array/);
});
