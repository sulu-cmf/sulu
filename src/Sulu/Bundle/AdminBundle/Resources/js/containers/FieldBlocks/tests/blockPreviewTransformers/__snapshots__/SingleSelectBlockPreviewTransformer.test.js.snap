// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Return JSX for selected item 1`] = `
<p>
  Value 1
</p>
`;

exports[`Return null if nothing is passed 1`] = `null`;
